lib/note-content-editor.jsx

```diff
@@ -0,0 +1,68 @@
+import React, { PropTypes } from 'react';
+import { Editor, EditorState, ContentState } from 'draft-js';
+import { urlDecorator } from './url-decorator';
+
+function plainTextContent( editorState ) {
+	return editorState.getCurrentContent().getPlainText( '\n' )
+}
+
+export default React.createClass( {
+
+	propTypes: {
+		content: PropTypes.string.isRequired,
+		onChangeContent: PropTypes.func.isRequired
+	},
+
+	getInitialState() {
+		const { content } = this.props;
+		const contentState = ContentState.createFromText( content, '\n' );
+		return {
+			editorState: EditorState.createWithContent( contentState, urlDecorator )
+		}
+	},
+
+	handleEditorStateChange( editorState ) {
+		const contentHasChanged = plainTextContent( editorState )
+			!== plainTextContent( this.state.editorState )
```

Comment:
@nfcampos could you surround these in parentheses and line up the args a bit? this is just a silly styling thing but I would like to avoid the hanging equality comparison. Alternatively we could use a couple of helper variables and make it short enough on a single line…

```js
const nextContent = plainTextContent( editorState );
const prevContent = plainTextContent( this.state.editorState );

const hasChangedContent = nextContent !== prevContent;
```